server/src/main/java/io/envoyproxy/controlplane/server/XdsDiscoveryRequestStreamObserver.java

```diff
@@ -0,0 +1,68 @@
+package io.envoyproxy.controlplane.server;
+
+import io.envoyproxy.controlplane.cache.Watch;
+import io.envoyproxy.envoy.api.v2.DiscoveryRequest;
+import io.envoyproxy.envoy.api.v2.DiscoveryResponse;
+import io.grpc.stub.StreamObserver;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+public class XdsDiscoveryRequestStreamObserver extends DiscoveryRequestStreamObserver {
+  private volatile Watch watch;
+  private volatile DiscoveryResponse latestResponse;
+  private Set<String> ackedResources;
```

Comment:
does this also need to be volatile?

Reply:
it shouldn't be necessarry because `ackedResources` is only used in the same flow on the same thread [here](https://github.com/envoyproxy/java-control-plane/pull/95/files/dbaa3f8724b7c99984030e5e490a160f94433569#diff-f3e46acc66b5d265516555f96bc02d1cR102) and [here](https://github.com/envoyproxy/java-control-plane/pull/95/files/dbaa3f8724b7c99984030e5e490a160f94433569#diff-f3e46acc66b5d265516555f96bc02d1cR108), should be safe to leave it as a normal field.

Reply:
Ok that makes sense. Mind adding a comment to that effect to help future readers?